nose2/plugins/mp.py

```diff
@@ -21,12 +21,31 @@ class MultiProcess(events.Plugin):
     def __init__(self):
         self.addArgument(self.setProcs, 'N', 'processes', '# o procs')
         self.testRunTimeout = self.config.as_float('test-run-timeout', 60.0)
-        self.procs = self.config.as_int(
-            'processes', multiprocessing.cpu_count())
+        self._procs = self.config.as_int(
+            'processes', 0)
         self.setAddress(self.config.as_str('bind_address', None))
 
         self.cases = {}
 
+    @property
+    def procs(self):
+        """Get the appropriate number of procs for self.procs if self._procs is
+        0."""
+
+        if self._procs == 0:
+            try:
+                self._procs = multiprocessing.cpu_count()
+            except NotImplementedError as e:
+                self._procs = 1
+        return self._procs
+
+    @procs.setter
+    def procs(self, value):
+        """Setter for procs property"""
+        if value <= 0:
```

Comment:
If you do `nose2 --plugin nose2.plugins.mp -N`  this will throw an `AttributeError`.   I would say this should be `< 0`, so that the default works.  And probably adjust the error message.    I probably should write a test case for this scenario, we can just file a bug for that.  Otherwise, I think the patch is fine.